src/Microsoft.Health.Fhir.Core/Features/Conformance/CapabilityStatementBuilder.cs

```diff
@@ -78,6 +78,13 @@ public ICapabilityStatementBuilder UpdateRestResourceComponent(string resourceTy
                 resourceComponent = new ListedResourceComponent
                 {
                     Type = resourceType,
+                    Profile = new ReferenceComponentImpl()
+                    {
+                        RefComponent = new ReferenceComponent()
+                        {
+                            Reference = $"http://hl7.org/fhir/StructureDefinition/{resourceType}",
+                        },
+                    },
```

Comment:
Not sure we need the nested structure, perhaps we can simplify to something like this?
```
Profile = new ReferenceComponent
{
  Reference = $"http://hl7.org/fhir/StructureDefinition/{resourceType}",
},
```

Reply:
I used the nested structure to retrieve the required values to serialize which I was not able to get without nested structure in the VersionSpecificReferenceConverter

if (value is IReferenceComponent obj)
            {
                if (_modelInfoProvider.Version.Equals(FhirSpecification.Stu3))
                {
                    serializer.Serialize(writer, obj.ReferenceComponent);
                }
                else
                {
                    serializer.Serialize(writer, obj.ReferenceComponent.Reference);
                }
            }

Reply:
Can you just use the object directly?
```csharp
if (value is ReferenceComponent obj)
{
    if (_modelInfoProvider.Version.Equals(FhirSpecification.Stu3))
    {
        serializer.Serialize(writer, obj);
    }
    else
    {
        serializer.Serialize(writer, obj.Reference);
    }
}
```

Then maybe rename to `ProfileReference` as we're assigning some implicit meaning to it

Reply:
I tried that initially and for stu3, it ran into JsonSerializationException: Self referencing loop detected with type. I can retry

Reply:
Got it! Lets sync in the morning, pretty sure we can fix that